moveos/moveos-types/src/filter.rs

```diff
@@ -0,0 +1,142 @@
+//! Blockchain filter
+
+use starcoin_vm_types::language_storage::type_tag_match;
+
+use crate::account_address::AccountAddress;
+use crate::block::BlockNumber;
+use crate::contract_event::ContractEvent;
+use crate::event::EventKey;
+use crate::language_storage::TypeTag;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Filter {
+    /// Blockchain will be searched from this block.
+    pub from_block: BlockNumber,
+    /// Till this block.
+    pub to_block: BlockNumber,
```

Comment:
The executor does not know the block, so this should be `from_tx` `to_tx`, But we can wait for the tx API to ensure this.